### ISSUE TITLE
Leaking credentials in log files

### ISSUE THREAD

**Issue by robin13** [logstash-plugins/logstash-output-elasticsearch#503]
- Version: 5.0.0
- Operating System: Ubuntu 16.04
- Config File (if you have sensitive info, please remove it):

```
input {
    generator {
        lines => [ "Foo Bar" ]
        count => 10
    }
}

output {
    elasticsearch {
        index => 'foo'
        user => 'my_name'
        password => 'my_password'
        hosts => [ 'localhost:9200']
    }
}
```

- Steps to Reproduce:

Run Logstash with user/password defined in the configuration, and the credentials are leaked in the log file:

```
[2016-11-11T11:36:42,596][INFO ][logstash.outputs.elasticsearch] Elasticsearch pool URLs updated {:changes=>{:removed=>[], :added=>["http://my_name:my_password@localhost:9200"]}}
```

Password should not be leaked as the permissions for log files may differ greatly from the configuration files.

**Comment by andrewvc** [comment]
Thanks! I need to pick up work on #450 which will eliminate this as a class of bugs.

**Comment by suyograo** [comment]
Fixed in #504
